.github/workflows/ci.yaml

```diff
@@ -0,0 +1,54 @@
+name: Build snap and run tests
+
+on:
+  workflow_call
+
+jobs:
+  check:
+    name: Code checks
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    strategy:
+      fail-fast: true
+      matrix:
+        python-version:
+          - '3.9'
+          - '3.10'
+    steps:
+      - id: checkout
+        name: Checkout repo
+        uses: actions/checkout@v3
+      - id: setup-python
+        name: Setup Python
+        uses: actions/setup-python@v2.2.2
+        with:
+          python-version: ${{matrix.python-version}}
+          architecture: x64
+      - id: code-checks
+        name: Make sure the code follows conventions and tests correctly
+        run: |
+          make checks
+
+  test:
+    name: Unit and Integration Tests
+    runs-on: ubuntu-latest
+    timeout-minutes: 60
+    needs:
+      - check
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
+      - id: setup-python
+        name: Setup Python
+        uses: actions/setup-python@v2.2.2
+        with:
+          python-version: ${{matrix.python-version}}
+          architecture: x64
+      - id: tests-unit
+        name: Run Unittests
+        run: |
+          make unittest
```

Comment:
All right, it bypassed my attention indeed -- I didn't notice that the `make check` command is running the unittests.
TBH I find this a bit un-intuitive...
Shouldn't the `checks` run checks (linting, bandit, whatsoever) and `make [unit|integration]test` phase run the tests?
I honestly think this should be split on the level of the `Makefile` and the Gitlab workflows should be able to refer to each step clearly.

Would you be OK with that solution?

Reply:
yes, I agree. If it is not too much, just add a Makefile rule with 

```
lint:
    tox -e lint

unittests:
    tox -e unit
```

or something along this line, such that this can be handled

Reply:
The `unittests` section is already there. What I was thinking to do was to be specific on `checks`.
Currently it's just calling `tox` with no parameter, thus implicitly using `tox` defaults (`lint, unit`).
I thought I could specifically pin it down here, independent from `tox` defaults such as:

```
 78 $(checks_tag): $(setup_tag)
 79     ${PYTHON} tox -e lint                                                                                                                                                                                       
 80     touch $(checks_tag)
 81 
 82 checks: $(checks_tag)
```
...or alternatively its better to consider `tox` defauls as the anytime "checks" we just ever wanna have? (In which case perhaps it's the `tox` defaults to modify :-) )

Reply:
I'm really fine with either, with possibly a slight preference to the first option to default to run both lint and unit with `tox`

Reply:
Yep, I agree